controllers/modelmesh.go

```diff
@@ -2,98 +2,114 @@ package controllers
-	if err := r.List(ctx, runtimes, client.InNamespace(namespace)); err != nil {
-		return err
-	}
-
-	// Build the payload processor endpoint
-	url := "http://" + crName + "." + namespace + ".svc.cluster.local/consumer/kserve/v2"
-
-	// Process ServingRuntimes
-	for _, runtime := range runtimes.Items {
-		err := processRuntime(ctx, r, &runtime, containerName, envVarName, url, remove)
-		if err != nil {
-			return err
+		// Check if all Pods are ready
+		if deployment.Status.ReadyReplicas != *deployment.Spec.Replicas {
+			// Not all replicas are ready, return an error
+			return false, fmt.Errorf("not all ModelMesh serving replicas are ready for deployment %s", deployment.Name)
 		}
-	}
-
-	return nil
-}
 
-func processRuntime(ctx context.Context, r client.Client, runtime *kserveapi.ServingRuntime, containerName string, envVarName string, url string, remove bool) error {
-
-	// Loop over the containers
-	for i := range runtime.Spec.Containers {
-		container := &runtime.Spec.Containers[i]
-
-		// Only update the specified container
-		if container.Name == containerName {
+		// Loop over all containers in the Deployment's Pod template
+		for i, _ := range deployment.Spec.Template.Spec.Containers {
 			// Get the existing env var
 			var envVar *corev1.EnvVar
-			for i, e := range container.Env {
+			for j, e := range deployment.Spec.Template.Spec.Containers[i].Env {
 				if e.Name == envVarName {
-					envVar = &container.Env[i]
+					envVar = &deployment.Spec.Template.Spec.Containers[i].Env[j]
 					break
 				}
 			}
 
 			// If the env var doesn't exist, add it (if we are not removing)
-			if envVar == nil {
-				if !remove {
-					container.Env = append(container.Env, corev1.EnvVar{
-						Name:  envVarName,
-						Value: url,
-					})
-				}
-			} else {
-				// Split the existing value into parts
-				parts := strings.Split(envVar.Value, " ")
-
-				// Check if the payload processor already exists in the list
-				exists := false
-				for _, part := range parts {
-					if part == url {
-						exists = true
-						break
+			if envVar == nil && !remove {
+				deployment.Spec.Template.Spec.Containers[i].Env = append(deployment.Spec.Template.Spec.Containers[i].Env, corev1.EnvVar{
+					Name:  envVarName,
+					Value: url,
+				})
+			} else if envVar != nil {
+				// If the env var exists and already contains the value, don't do anything
+				existingValues := strings.Split(envVar.Value, " ")
+				for _, v := range existingValues {
+					if v == url {
+						continue
 					}
 				}
 
-				// If we are removing, and the payload processor exists, remove it
-				if remove && exists {
-					newParts := []string{}
-					for _, part := range parts {
-						if part != url {
-							newParts = append(newParts, part)
-						}
-					}
-					envVar.Value = strings.Join(newParts, " ")
-				}
+				// Modify the existing env var based on the remove flag and current value
+				envVar.Value = updateEnvVarValue(envVar.Value, url, remove)
```

Comment:
Where this value is updated? Is it line 53 correct?

Reply:
Yes, the last step of the loop.